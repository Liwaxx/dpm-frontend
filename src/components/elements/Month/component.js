import React from 'react';
import PropTypes from 'prop-types';
import { TableRow, TableCell } from '@material-ui/core';
import { IMAGES } from '../../../configs';

export default class Component extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      count: 2018
    };
  }

  handleIncrement() {
    this.setState({
      count: this.state.count + 1
    });
  }

  handleDecrement() {
    this.setState({
      count: this.state.count - 1
    });
  }

  render() {
    const { classes } = this.props;
    return (
      <TableRow className={classes.line}>
        <TableCell>
          <button onClick={() => this.handleDecrement()} className={classes.left}>
            <img src={IMAGES.ARROW_LEFT} />
          </button>
        </TableCell>
        <TableCell>
          <div className={classes.tahun}>
            {this.state.count}
          </div>
        </TableCell>
        <TableCell>
          <button onClick={() => this.handleIncrement()} className={classes.right}>
            <img src={IMAGES.ARROW_RIGHT} />
          </button>
        </TableCell>
      </TableRow>
    );
  }
}

Component.propTypes = {
  classes: PropTypes.object.isRequired,
};