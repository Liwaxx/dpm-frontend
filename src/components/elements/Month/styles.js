const styles =  ({
  left: {
    border: 'none',
    backgroundColor: '#fff',
    marginLeft: '10px',
  },
  right: {
    border: 'none',
    backgroundColor: '#fff',
    marginLeft: '-80px',
  },
  tahun: {
    fontFamily: 'roboto',
    marginLeft: '-60px',
    fontSize: '20px',
  },
  line: {
    width: '180px',
  }
});
export default styles;