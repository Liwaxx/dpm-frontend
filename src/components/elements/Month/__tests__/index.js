import React from 'react';
import renderer from 'react-test-renderer';
import Month from '../index';

jest.mock('../../../../configs');

describe('Month', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Month />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});