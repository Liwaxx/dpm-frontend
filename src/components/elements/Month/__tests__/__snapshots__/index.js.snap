// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Month renders correctly 1`] = `
<tr
  className="MuiTableRow-root-5 Component-line-4"
>
  <td
    aria-sort={null}
    className="MuiTableCell-root-10"
  >
    <button
      className="Component-left-1"
      onClick={[Function]}
    >
      <img />
    </button>
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root-10"
  >
    <div
      className="Component-tahun-3"
    >
      2018
    </div>
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root-10"
  >
    <button
      className="Component-right-2"
      onClick={[Function]}
    >
      <img />
    </button>
  </td>
</tr>
`;
